src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -1749,12 +1736,18 @@ class MatrixTable(val hc: HailContext, val metadata: VSMMetadata,
     val localNSamples = nSamples
     val localSampleIdsBc = sampleIdsBc
     val localSampleAnnotationsBc = sampleAnnotationsBc
+    val localRowType = rowType
 
     Table(hc,
-      rdd.mapPartitions { it =>
+      rdd2.mapPartitions { it =>
         val n = vNames.length + gNames.length * localNSamples
 
-        it.map { case (v, (va, gs)) =>
+        it.map { rv =>
+          val ur = new UnsafeRow(localRowType, rv)
```

Comment:
I think you can move the `UnsafeRow` constructor outside of the `it.map` if you want, here.